pkg/cloudprovider/aws/amifamily/bootstrap/eksbootstrap.go

```diff
@@ -40,16 +41,30 @@ func (e EKS) Script() string {
 
 	kubeletExtraArgs := strings.Join([]string{e.nodeLabelArg(), e.nodeTaintArg()}, " ")
 
-	if !e.AWSENILimitedPodDensity {
-		userData.WriteString(" \\\n--use-max-pods=false")
-		kubeletExtraArgs += " --max-pods=110"
+	clusterDNSIP := e.Options.KubeDnsIP
+
+	if e.KubeletConfig != nil && len(e.KubeletConfig.ClusterDNS) > 0 {
+		clusterDNSIP = net.ParseIP(e.KubeletConfig.ClusterDNS[0])
+		userData.WriteString(fmt.Sprintf(" \\\n--dns-cluster-ip '%s'", clusterDNSIP))
+	}
+
+	ipv6Cluster := isIPv6(clusterDNSIP)
+	if ipv6Cluster {
+		userData.WriteString(" \\\n--ip-family ipv6")
+		_, ipNet, err := net.ParseCIDR(clusterDNSIP.String() + "/108")
+		if err != nil {
+			panic(fmt.Sprintf("Parsing IPv6 cluster DNS, %s", err))
+		}
+		userData.WriteString(fmt.Sprintf(" \\\n--service-ipv6-cidr '%s'", ipNet))
+	}
+
+	if !e.AWSENILimitedPodDensity || ipv6Cluster {
+		userData.WriteString(" \\\n--use-max-pods false")
+		kubeletExtraArgs += " --max-pods 110"
```

Comment:
Comment: `--max-pods=110` is the default, so it's fine to leave it here for clarity - but we could also omit it and it would have the same result.